python/graphstorm/dataloading/dataset.py

```diff
@@ -446,15 +447,28 @@ def prepare_data(self, g):
                 if dist_sum(len(test_idx)) > 0:
                     test_idxs[canonical_etype] = test_idx
             else:
-                print(f"WARNING: {canonical_etype} does not contains " \
-                      "test_mask, skip testing {canonical_etype}")
+                print(f"NOTE: {canonical_etype} does not contains " \
+                      "test_mask, skip testing {canonical_etype}. \n" \
+                      "We will do inference on the whole edge list.")
+                infer_idx = dgl.distributed.edge_split(
+                    th.full((g.num_edges(canonical_etype),), True, dtype=th.bool),
+                    pb, etype=canonical_etype, force_even=True)
+                infer_idxs[canonical_etype] = infer_idx
+
         self._test_idxs = test_idxs
+        self._infer_idxs = infer_idxs
 
     @property
     def eval_etypes(self):
         """edge type for evaluation"""
         return self._eval_etypes
 
+    @property
+    def infer_idxs(self):
```

Comment:
i don't understand the logic here. how is `infer_idxs` different from `test_idxs`? it seems to me that if test edges are not provided, we compute scores over all edges?

Reply:
You are right. If there is no test edges, we compute scores for all edges.

Reply:
But for infer edges, no label is needed.

Reply:
what does infer edges mean? do you mean to compute the scores on all edges? this is link prediction. what do you mean by labels?

Reply:
Edge idxs for inference. This dataset can also be used by edge classificaiton/regression.

Reply:
i see. i checked `GSgnnNodeInferData`. it doesn't have this method. why is this required for `GSgnnEdgeInferData`?

Reply:
If we want to support node inference without test_mask, we also need to add infer_idxs in GSgnnNodeInferData.
The reason I add infer_idxs instead of re-using test_idx is because, I want to separate this case with the case that has test_mask and will output test scores.

Reply:
I will add inference without test mask support for GSgnnNodeInferData in another PR.

Reply:
@zheng-da Any concerns?